ehrapy/preprocessing/_encoding.py

```diff
@@ -628,72 +567,30 @@ def _reorder_encodings(adata: AnnData, new_encodings: dict[str, list[list[str]]
-            for column_list in encoded_categoricals_with_mode:
-                for column_name in column_list:
-                    if column_name == categorical:
-                        column_list.remove(column_name)
-                        _found = True
-                        break
-                # a categorical can only be encoded once and therefore found once
-                if _found:
-                    break
-            # filter all lists that are now empty since all variables will be reencoded from this list
-            updated_multi_list = filter(None, encoded_categoricals_with_mode)
-            # if no columns remain that will be encoded with this encode mode, delete this mode from modes as well
-            if not list(updated_multi_list):
-                del adata.uns["encoding_to_var"][encode_mode]
-        # single column encoding mode
-        elif encode_mode in available_encodings:
-            encoded_categoricals_with_mode = adata.uns["encoding_to_var"][encode_mode]
-            for ind, column_name in enumerate(encoded_categoricals_with_mode):
-                if column_name == categorical:
-                    del encoded_categoricals_with_mode[ind]
-                    break
-                # if encoding mode is
-            if not encoded_categoricals_with_mode:
-                del adata.uns["encoding_to_var"][encode_mode]
-
-    return _update_new_encode_modes(new_encodings, adata.uns["encoding_to_var"])
-
-
-def _update_new_encode_modes(
-    new_encodings: dict[str, list[list[str]] | list[str]],
-    filtered_old_encodings: dict[str, list[list[str]] | list[str]],
-):
-    """Update the encoding scheme.
-
-    If the encoding mode exists in the filtered old encodings, append all values (columns that should be encoded using this mode) to this key.
-    If not, defaultdict ensures that no KeyError will be raised and the values are simply appended to the default value ([]).
 
-    Args:
-        new_encodings: The new encoding modes passed by the user; basically what will be passed for encodings when calling the encode API
-        filtered_old_encodings: The old encoding modes, but with all columns removed that will be reencoded
+    encodings = {}
+    for encode_mode in available_encodings:
+        encoded_categoricals_with_mode = (
+            adata.var.loc[adata.var["encoding_mode"] == encode_mode, "unencoded_var_names"].unique().tolist()
+        )
 
-    Returns:
-        The updated encoding scheme
-    """
-    updated_encodings = defaultdict(list)  # type: ignore
-    for k, v in chain(new_encodings.items(), filtered_old_encodings.items()):
-        updated_encodings[k] += v
+        encodings[encode_mode] = new_encodings[encode_mode] if encode_mode in new_encodings.keys() else []
+        # add all columns that were encoded with the current mode before and are not reencoded
+        encodings[encode_mode] += [cat for cat in encoded_categoricals_with_mode if cat not in latest_encoded_columns]
```

Comment:
![image](https://github.com/theislab/ehrapy/assets/21954664/40ca55d4-b0a3-4426-9fd1-dbfdff48efd9)

Reply:
Changed the variable name to `categorical`^^ No more cats...🥲